### PR TITLE
Resizable Nodes and Services table columns

### PR DESCRIPTION
**Pull request by mperrotti**
## Testing
(recommended in a private browsing tab)
1. Navigate to the {Nodes|Services} page
2. Resize some columns
3. Refresh the page

Columns should be resizable, and when you refresh the page, the resized columns should be the same width you resized them to

## Screenshots
ServicesResizeDemo

### CODE REVIEW

File: plugins/nodes/src/js/components/NodesTable.tsx [hunk 1]
```diff
@@ -297,8 +439,21 @@ export default class NodesTable extends React.Component<
               />
             }
             cellRenderer={memRenderer}
-            growToFill={true}
-            minWidth={110}
+            growToFill={
```
Comment (jaxesn) [inline plugins/nodes/src/js/components/NodesTable.tsx]: This is all fine for this PR, but I wonder if there could be a future improvement to how we handle these resizeable columns...  Having the Store to store this state makes a lot of sense so having this kind of logic in dcos-ui vs ui-kit is reasonable.  Alot of this PR, esp the lines following, are boilerplate-ish and some of potentially error-proned, like this growToFill "logic"...  Would there be a slick way to have a HOC or some other kind of component/helper in dcos-ui that can almost "generate" this boilerplate to make the ultimate api something like:

<Column minWitdth={110} resizeable={true} columnKey={"mem"} />

**Comment by GeorgiSTodorov** [comment]
Also, unit and snapshot tests are failing.
Screenshot from 2019-04-11 11-54-31

**Comment by GeorgiSTodorov** [comment]
Currently looking into the non-snapshot failure.

**Comment by GeorgiSTodorov** [comment]
Merging master to see if it will fix the issue.

**Comment by GeorgiSTodorov** [comment]
The same test is failing again.
Screenshot from 2019-04-11 13-58-01

File: plugins/nodes/src/js/components/NodesTable.tsx (hunk 1 above)
Comment (nLight) [reply plugins/nodes/src/js/components/NodesTable.tsx]: I'd agree with Jackson. We should design the interface a little here.

```jsx
<ConfigurableTable id="table_id" storeBackend={LocalStorageBackend}>
    <ResizableColumn minWitdth={110} columnKey="mem" />
</ConfigurableTable>
```

or even

```jsx
<Configuration id="table_id" storeBackend={LocalStorageBackend}>
  <Table>
    <ResizableColumn minWitdth={110} columnKey="mem" />
  </Table>
</Configuration>
```

Configuration would inject defaults into its child component. We would need figuring out the `onConfigurationChange` interface though.

**Comment by GeorgiSTodorov** [comment]
Taking a look at the failing tests.

**Comment by GeorgiSTodorov** [comment]
I've somewhat narrowed down the problem. It comes from the new ui-kit version. It makes the table have height 0 in integration tests.
I tested this by first running master and seeing tests work and then running master with new ui-kit version 2.1.1 and seeing tests break because of invisible tables.

**Comment by mperrotti** [comment]
This ui-kit PR should fix the issue: https://github.com/dcos-labs/ui-kit/pull/324

Alternatively, we could have added styles to ensure the `div` that is a direct child of `.table-wrapper` fills it's parent's height. The ui-kit fix feels much cleaner though

**Comment by GeorgiSTodorov** [comment]
If there are still failing tests, I will investigate them.

**Comment by GeorgiSTodorov** [comment]
Can't reproduce failing tests locally, retriggered.

**Comment by GeorgiSTodorov** [comment]
Still can't reproduce them locally. Reasons for test failures looking at the artifact video:
Screenshot from 2019-04-18 12-13-20
Screenshot from 2019-04-18 12-14-48

It looks like a flake to me.

**Comment by mperrotti** [comment]
@GeorgiSTodorov - I wasn't able to reproduce locally either. I wonder why these tests aren't flaking on other PRs though 🤔

**Comment by GeorgiSTodorov** [comment]
@mperrotti , are we sure they aren't?

**Comment by GeorgiSTodorov** [comment]
Merging latest master to see if they are still flakes.

**Comment by GeorgiSTodorov** [comment]
Now it fails locally.
Screenshot from 2019-04-25 14-50-36

File: tests/pages/services/ServiceFormModal-cy.js [hunk 2]
```diff
@@ -500,6 +500,7 @@ describe("Service Form Modal", function() {
             .focus()
             .type("{selectall}{backspace}")
             .type("{selectall}{backspace}")
+            .type("{selectall}{backspace}")
```
Comment (mperrotti) [inline tests/pages/services/ServiceFormModal-cy.js]: Locally, this prevents Cypress from typing part of the input at the beginning of the field. This is more of a workaround than an actual fix, but I want to see if it works in CI

**Comment by nLight** [comment]
flicts :)

**Pull request merged**